sky/templates/aws-ray.yml.j2

```diff
@@ -25,8 +25,13 @@ provider:
   # teardown(terminate=True) will override this.
   cache_stopped_nodes: True
   security_group:
-    # AWS config file must include security group name
+    # AWS config file must include security group name, but we change
+    # to dedicted security group if we have ports requirement.
+  {% if have_ports %}
+    GroupName: sky-sg-{{cluster_name_on_cloud}}
+  {% else %}
     GroupName: {{security_group}}
+  {% endif %}
```

Comment:
Why don't we have the `clouds.AWS.make_deploy_variables` to handle this

Reply:
Because resource don't have the information of `cluster_name_on_cloud`

Reply:
I think it should be fine to add the `cluster_name_on_cloud` API to the `make_deploy_variables`. Reason: the current way to replace the name in ray yaml can be very hard to track, as we logic for deciding the sg name are separated in many different places.

Reply:
Done ;)